### PR TITLE
Jetpack: Add Salesforce Lead form template

### LINKED ISSUE #26868: Jetpack: Contact From -Integrate Salesforce for sending leads
A simple implementation that just sends leads to a Salesforce organization

### PR DESCRIPTION
**Pull request by CGastrell**
#### Changes proposed in this Pull Request:
This PR adds a new form template: Salesforce Lead Form. The form is a variation of the contact-form with a fixed set of inputs:
- Email
- First name
- Last name
- Job/Title
- Company
- Phone number

Sidebar settings include the Organization ID and (optional) a Campaign ID

Closes #26868 

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
The change is being discussed/tracked at pcsBup-dV-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:
You'll need a Salesforce developer account, you can get one for free at salesforce.com.
Once you do, use the Jurassic Ninja link to create a new instance with this PR.
Change the Jetpack Constant JETPACK_BETA_BLOCKS to `true` by visiting the Settings->Jetpack Constants entry on WP-ADMIN

**NOTE: do NOT check JETPACK_EXPERIMENTAL_BLOCKS as it will override the BETA list. If it's checked, uncheck it.**

Edit a post and add a `/form`.

- [ ] you should see a new entry: Salesforce Lead Form

Add a Salesforce Lead Form.

- [ ] mentioned list of inputs should be present
- [ ] settings on the sidebar should allow you to enter an Organization ID

Follow the instructions on the settings to get your Organization ID, enter it on the settings
Visit the post and fill the form. Then, on your Salesforce instance, go to Apps -> Marketing. You should see a "Leads" tab.

- [ ] verify a lead now exists with the data you filled the form with

### CODE REVIEW

**Comment by digitalwaveride** [comment]
Very cool Christian!

I think we can for now hide the "Campaign ID" section for the discussed reasons that it is complicated for Salesforce users to access these, and our assumption that users want to do their filtering and "routing" of incoming leads anyways on the Salesforce side, as there is much more functionality.

Regarding the placeholder screen, it would be great to get the new additional template also into the first line.

Currently:
image

Better:
image

File: projects/plugins/jetpack/extensions/blocks/contact-form/edit.js
Comment (jcheringer) [inline projects/plugins/jetpack/extensions/blocks/contact-form/edit.js]: Shouldn't we move this logic to the `variations.js` file?
By doing that, we ensure that wherever the `variations` are used, it'll contain all existent variations.
Right now, we can see the Salesforce variation on the placeholder, but if you search in the Block Library, you won't find it.
That's because the variations we see in the Library are registered by the `index.js` file.
Also, if we move this logic, we could reutilize the `defaultBlockStyling` introduced by you a few days ago.

**Review by jcheringer — CHANGES_REQUESTED** [review]
Worked well for me. I'm just adding a comment about making the Salesforce variation searchable in the Block Library

File: projects/plugins/jetpack/extensions/blocks/contact-form/edit.js
Comment (CGastrell) [reply projects/plugins/jetpack/extensions/blocks/contact-form/edit.js]: Yes to all of the above, but that would also involve the feature flag to be read on more than one place. Right now I want this hidden for internal testing, and this check makes for a single point of entry for the variation. Once we get this reviewed and tested, we'd remove the flag and make this into the main variations file, alongside the defaultBlockStyling optimization.

I'm not sure it's the best way, but it made sense to me while working in it. Let me know if you think otherwise.

Comment (jcheringer) [reply projects/plugins/jetpack/extensions/blocks/contact-form/edit.js]: If not having the Salesforce template available in the Library isn't an issue now, I'll give you the ✅.
But I'll leave it to @digitalwaveride to decide.

**Review by jcheringer — DISMISSED** [review]

File: projects/plugins/jetpack/3rd-party/class-salesforce-lead-form.php
Comment (zinigor) [inline projects/plugins/jetpack/3rd-party/class-salesforce-lead-form.php:83]: Wouldn't we also get this message as a fatal error if we just don't catch the thrown exception? Were you planning to add a handler of some sort here later?

**Review by zinigor — DISMISSED** [review]
I have only one comment/question, but overall the PR looks fine to me.

File: projects/plugins/jetpack/3rd-party/class-salesforce-lead-form.php
Comment (CGastrell) [reply projects/plugins/jetpack/3rd-party/class-salesforce-lead-form.php:83]: Yes, but I think I can tackle this as we continue to work on customizations (next iteration expected as soon as this is merged). @zinigor do you have any tips regarding error logging (kibana or otherwise)?

**Review by digitalwaveride — DISMISSED** [review]
👍 Looks good to me!
Thanks Christian!

**Comment by CGastrell** [comment]
Hey @zinigor Thanks for the review! Sadly I had to merge trunk back into this as it got outdated.

Regarding the `trigger_error` call:

- right now, that seems like the _de facto_ way of invalidating a form submission
- that said, we'll continue to work on this on a second level of customization (inputs-to-salesforce-api-fields), where we can surely make for more consistent checks and invalidations
- the whole thing is still behind the beta flag, so we can work on it until it's ready for public

Does it make sense? Can I get a second review/approval?

**Comment by bindlegirl** [comment]
@CGastrell This looks good to me and since the feature is behind beta flag it makes sense to address the issue @zinigor mentioned in upcoming PRs.

I had to rebase the PR and resolved a conflict in edit.js in the process related to `const formClassnames` on line 78. ([PR that made the change](https://github.com/Automattic/jetpack/pull/27134)). As far as I can tell, everything looks OK now, but I would appreciate you check the code again to make sure I made the correct decision while resolving it.

**Review by bindlegirl — APPROVED** [review]

**Comment by CGastrell** [comment]
Thanks @bindlegirl ! Change is good (I think I've seen it before and I thought I had already rebased 🤷 )

**Pull request merged**
